### PR TITLE
IPL3 documentation; undefined_syms.txt finally deleted

### PR DESCRIPTION
**Pull request by someone2639**
-Documentation + use of real addresses in r4300.h and rcp.h
-added a quick and dirty checksum skip define check (but not the define itself)
-added a quick and dirty way to set how big the initial 1mb dma is, and where in the rom it starts

### CODE REVIEW

**Review by gheskett — APPROVED** [review]
I've got no idea what's going on here, but I'll approve anyway to keep this PR from being permanently added to the pile. Other reviews are encouraged if people can make sense out of this.

File: asm/boot.s
Comment (someone2639) [inline asm/boot.s:455]: This part actually checks the checksum found in the ROM with the checksum it calculated; you can just remove this check and it still works on mupen even if you mess with the checksum in a hex editor (not integrated into config.h purely because no idea how other emulators deal with it)

File: include/n64/PR/ultratypes.h
Comment (someone2639) [inline include/n64/PR/ultratypes.h:36]: Need this guard so the assembler doesnt get caught up in typedefs

**Review by someone2639 — COMMENTED** [review]
comments for reviewers

**Comment by Reonu** [comment]
yip file

**Review by thecozies — CHANGES_REQUESTED** [review]
do you think you could rename boot.s to `ipl3.s` so its a bit more accurate to what it do?

**Comment by thecozies** [comment]
after rename it has my "ping me for official approval" approval

**Comment by CrashOveride95** [comment]
Editing IPL3 doesn't seem safe? That will obviously change the hash which may potentially break things, has this been tested on real hardware?

**Comment by someone2639** [comment]
addressed

**Review by thecozies — APPROVED** [review]

**Pull request merged**
